src/ssl.cpp

```diff
@@ -792,48 +792,68 @@ int SSL_CTX_load_verify_locations(SSL_CTX* ctx, const char* file,
         WIN32_FIND_DATA FindFileData;
         HANDLE hFind;
 
-        char name[MAX_PATH + 1];  // directory specification
-        strncpy(name, path, MAX_PATH - 3);
+        char* name = NEW_YS char[strlen(path) + 4];  // directory specification
+        int nameSz = strlen(path) + 4; // 4 for extra space needed with string
+        strncpy(name, path, nameSz - 3);
         strncat(name, "\\*", 3);
 
         hFind = FindFirstFile(name, &FindFileData);
-        if (hFind == INVALID_HANDLE_VALUE) return SSL_BAD_PATH;
+        if (hFind == INVALID_HANDLE_VALUE) {
+            ysArrayDelete(name);
+            return SSL_BAD_PATH;
+        }
 
         do {
-            if (FindFileData.dwFileAttributes != FILE_ATTRIBUTE_DIRECTORY) {
-                strncpy(name, path, MAX_PATH - 2 - HALF_PATH);
+            if (!(FindFileData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)) {
+                if ((int)strlen(path) + 4 +
+                                 (int)strlen(FindFileData.cFileName) > nameSz) {
+                    ysArrayDelete(name);
+                    nameSz = strlen(path) + 4 + strlen(FindFileData.cFileName);
+                    name = NEW_YS char[nameSz];
+                }
+                memset(name, 0, nameSz);
+                strncpy(name, path, nameSz);
```

Comment:
While path len is less than nameSz, a strncpy of full buffer size could leave an unterminated string which is a bug.